library/spdm_crypt_lib/libspdm_crypt_asym.c

```diff
@@ -833,28 +940,97 @@ bool libspdm_asym_verify_hash(
             return libspdm_asym_verify_wrap(context, hash_nid, base_asym_algo,
                                             param, param_size,
                                             full_message_hash, hash_size,
-                                            signature, sig_size);
+                                            signature, sig_size,
+                                            false);
         } else {
             return libspdm_asym_verify_wrap(context, hash_nid, base_asym_algo,
                                             param, param_size,
                                             message, message_size,
-                                            signature, sig_size);
+                                            signature, sig_size,
+                                            false);
         }
 
         /* SPDM 1.2 signing done. */
     }
 
     if (need_hash) {
+        try_endian_swapped_signature
+            = libspdm_try_endian_swapped_signature(spdm_version, base_asym_algo);
+
         return libspdm_asym_verify_wrap(context, hash_nid, base_asym_algo,
                                         param, param_size,
                                         message_hash, hash_size,
-                                        signature, sig_size);
+                                        signature, sig_size,
+                                        try_endian_swapped_signature);
     } else {
         LIBSPDM_ASSERT(false);
         return false;
     }
 }
 
+void libspdm_asym_signature_swap_endian_if_necessary(
+    spdm_version_number_t spdm_version, uint32_t base_asym_algo,
+    uint8_t *signature, size_t sig_size)
+{
+#if LIBSPDM_SPDM_10_11_SIGN_LITTLE_ENDIAN_RSA_ECDSA
+    uint32_t swap_endian_rsa = 0;
+    uint32_t swap_endian_ecdsa = 0;
```

Comment:
I think swap_endian_rsa and swap_endian_ecdsa should be a `bool` type.

Reply:
Agree.